tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -656,11 +657,16 @@ static __always_inline u32 get_task_ns_pid(struct task_struct *task)
     struct pid_namespace *pid_ns_children = READ_KERN(namespaceproxy->pid_ns_for_children);
     unsigned int level = READ_KERN(pid_ns_children->level);
 
+#ifndef CORE
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 19, 0) && !defined(RHEL_RELEASE_GT_8_0))
-    // kernel 4.14-4.18:
+    // kernel 4.14-4.18
     return READ_KERN(READ_KERN(task->pids[PIDTYPE_PID].pid)->numbers[level].nr);
 #else
-    // kernel 4.19 onwards, and CO:RE:
+    // kernel 4.19 onwards
+    struct pid *tpid = READ_KERN(task->thread_pid);
+    return READ_KERN(tpid->numbers[level].nr);
+#endif
+#else // CORE
     struct pid *tpid = READ_KERN(task->thread_pid);
     return READ_KERN(tpid->numbers[level].nr);
 #endif
```

Comment:
Can we avoid code duplication here (reading thread_pid)?
The only place where we need to use the older task->pids is in kernel 4.18 (BTW we should remove the 4.14-4.18 comment).
By reading this code I also understand that we should probably use flavored struct for kernel 4.18.

Same comment for the next two functions.

Reply:
CO-RE is not working in kernels <= 5.0, that is why we only have 4.18 + (4.19 & CORE). I asked you if you had CORE tested in kernels < 5.0 in the issue.. have you ever tested CORE in 4.18 or 4.19 ? And, if not, should we consider CORE is only supported in >= 5.0 (working and tested).

Reply:
Nope. I didn't test CORE on kernels < 5.0, (only the non CO-RE version).
We should open an issue to check why it fails to run on those kernels

Reply:
And I propose to add the 4.18 flavor within this same issue. I opened the following issue to keep track of that: https://github.com/aquasecurity/tracee/issues/1171